### PR TITLE
Fixes #447 [Watson] clr20r3: CLR_EXCEPTION_System.ArgumentNullExcepti…

### LINKED ISSUE #447: [Watson] clr20r3: CLR_EXCEPTION_System.ArgumentNullException_80004003_Microsoft.PythonTools.dll!Microsoft.PythonTools.Repl.BasePythonReplEvaluator+CommandProcessorThread.SendString
Opened from bug #1184574 in TFS.

From Cab: 94517130568 

```
mscorlib_ni!System.Text.Encoding.GetBytes
microsoft_pythontools!Microsoft.PythonTools.Repl.BasePythonReplEvaluator+CommandProcessorThread.SendString
microsoft_pythontools!Microsoft.PythonTools.Repl.BasePythonReplEvaluator+CommandProcessorThread+__c__DisplayClass28._ExecuteFile_b__27
microsoft_pythontools!Microsoft.PythonTools.Repl.BasePythonReplEvaluator+CommandProcessorThread.ExecuteFile
microsoft_pythontools!Microsoft.PythonTools.Repl.BasePythonReplEvaluator.ExecuteFile
microsoft_pythontools!Microsoft.PythonTools.Commands.ExecuteInReplCommand+__c__DisplayClass1._DoCommand_b__0
mscorlib_ni!System.Threading.QueueUserWorkItemCallback.WaitCallback_Context
mscorlib_ni!System.Threading.ExecutionContext.RunInternal
mscorlib_ni!System.Threading.ExecutionContext.Run
mscorlib_ni!System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem
mscorlib_ni!System.Threading.ThreadPoolWorkQueue.Dispatch
mscorlib_ni!System.Threading._ThreadPoolWaitCallback.PerformWaitCallback
```

Other Cabs related to this Failure
These cabs may or may not have been deleted under the Watson retention policy.
Source Location 
Watson \\WTELFILES\WatsonCabs\Common\All\Cab03\451\713\94517130568.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab01\451\924\94519244017.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab02\452\094\94520943079.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab04\452\267\94522673221.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab04\451\744\94517445347.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab03\452\354\94523543349.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab03\452\379\94523792405.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab02\452\380\94523807341.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab04\452\847\94528479152.cab  
Watson \\WTELFILES\WatsonCabs\Common\All\Cab02\452\741\94527418908.cab

### PR DESCRIPTION
**Pull request by zooba**
Fixes #447 [Watson] clr20r3: CLR_EXCEPTION_System.ArgumentNullException_80004003_Microsoft.PythonTools.dll!Microsoft.PythonTools.Repl.BasePythonReplEvaluator+CommandProcessorThread.SendString
Ensures StartInfo.Arguments is always treated as if it may be null (which it may be).

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/Project/CustomCommand.cs [hunk 1]
```diff
@@ -861,9 +867,10 @@ class CommandStartInfo : IProjectLaunchProperties {
             } else if (ExecuteInConsole) {
                 if (handleConsoleAndPause) {
                     Arguments = string.Format(
-                        "/C \"{0} {1}\" & if errorlevel 1 pause",
+                        "/C \"{0}{1}{2}\" & if errorlevel 1 pause",
                         ProcessOutput.QuoteSingleArgument(Filename),
-                        Arguments
+                        string.IsNullOrEmpty(Arguments) ? "" : " ",
```
Comment (gilbertw) [inline Python/Product/PythonTools/PythonTools/Project/CustomCommand.cs]: String.Empty instead of ""

**Comment by gilbertw** [comment]
:+1:

File: Python/Product/PythonTools/PythonTools/Repl/BasePythonReplEvaluator.cs [hunk 2]
```diff
@@ -977,7 +977,8 @@ public DisconnectedException(string message)
             }
 
             private void SendString(string text) {
-                byte[] bytes = System.Text.Encoding.UTF8.GetBytes(text);
+                Debug.Assert(!string.IsNullOrEmpty(text), "text should not be null or empty");
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/Repl/BasePythonReplEvaluator.cs]: Are we calling this SendString up above in 959-962 with an empty string only to assert here?

Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Repl/BasePythonReplEvaluator.cs]: That should be a null check only. Empty strings are okay here

**Comment by huguesv** [comment]
:+1:

**Pull request merged**